### PR TITLE
Swift 3.2 compatibility

### PR DESCRIPTION
**Pull request by cristianames92**
## Issue ##
https://github.com/Wolox/wolmo-core-ios/issues/73

This brings support for Swift 3.2 using Xcode 9. Next step would be migrate to Swift 4

## Known issues ##
The ReactiveCocoa reference in the Cartfile is directly pointing to an alpha release, something that should be changed to the `6.0` series when they release a final version of the library.

### CODE REVIEW

File: CoreTests/Extensions/Foundation/BundleSpec.swift [hunk 1]
```diff
@@ -268,7 +268,7 @@ public class BundleSpec: QuickSpec {
                     
                     it("should return .none") {
                         let value = bundle.getInt(from: "My Personal Boolean Key")
-                        expect(value).to(beNil())
+                        expect(value).to(equal(1))
```
Comment (danielaRiesgo) [inline CoreTests/Extensions/Foundation/BundleSpec.swift]: Why should this change?

File: script/.env [hunk 2]
```diff
@@ -1,6 +1,6 @@
-REQUIRED_XCODE_VERSION=8.3
+REQUIRED_XCODE_VERSION=9.0
 REQUIRED_SWIFTLINT_VERSION=0.16.1
```
Comment (danielaRiesgo) [inline script/.env]: Isn't there an updated swiftlint version for swift 2.3 additions/changes?

**Comment by pablog** [comment]
@cristianames92

File: CoreTests/Extensions/Foundation/BundleSpec.swift (hunk 1 above)
Comment (cristianames92) [inline CoreTests/Extensions/Foundation/BundleSpec.swift]: It seems that the new changes involved this new interpretation of the boolean value. If you try to cast it to an int it will apparently return 0 or 1 according the bool value.

File: script/.env (hunk 2 above)
Comment (cristianames92) [inline script/.env]: Not yet, but Realm has more recent releases for the old swift 3.0 versions. They are now under the version 0.19.0. I didn't change it because there is no new version for Swift 3.2.

File: CoreTests/Extensions/Foundation/BundleSpec.swift (hunk 1 above)
Comment (danielaRiesgo) [inline CoreTests/Extensions/Foundation/BundleSpec.swift]: Okay, so then we should update the documentation and test description.

File: script/.env (hunk 2 above)
Comment (danielaRiesgo) [inline script/.env]: Okay. I would update it to 0.19.0 none the less.

File: CoreTests/Extensions/Foundation/BundleSpec.swift [hunk 3]
```diff
@@ -264,9 +264,9 @@ public class BundleSpec: QuickSpec {
                     
                 }
                 
-                context("when the value is not the expected type") {
+                context("when the value can be converted in other types") {
```
Comment (danielaRiesgo) [inline CoreTests/Extensions/Foundation/BundleSpec.swift]: Shouldn't we then add another test for the not expected type failing to None?

**Comment by cristianames92** [comment]
The project will still failing in Bitrise due to the version of XCode in the server

**Comment by pablog** [comment]
@damfinkel could you pelase check if bitrise is failing because of the same you found a workaround?

**Comment by danielaRiesgo** [comment]
I can't see the logs, they never end loading. @damfinkel any clue?

**Comment by damfinkel** [comment]
Closing this pull request, will migrate directly to swift 4
